components/sup/src/manager/service/mod.rs

```diff
@@ -62,6 +62,9 @@ pub use self::supervisor::ProcessState;
 
 static LOGKEY: &'static str = "SR";
 
+pub const SVC_DIR_PERMISSIONS: u32 = 0o770;
```

Comment:
I feel like `7` (`rwx`) is probably right for group here, although on first look I expected to see `5` (`rx`). Is this because the priv-dropped user may need to write under these directories? I might have answered my own question.

Oh while we're doing this, it might be worth considering adding a sticky bit to the group which would preserve the group of created child directories (I believe `0o1770` would do the trick). I don't think this is critical, but maybe worth a comment or `TODO` if it looks like this would be handy in the future?

Reply:
I actually did start this with lower permissions, but my `nginx` test case ended up complaining because it didn't have sufficient privileges along the entire directory path.

I'll look into the sticky bit solution.

Reply:
@fnichol Actually, can you give a bit more idea of what you're thinking re: sticky bit? I'm not seeing how it would come into play here.

Reply:
Sure! If we had a service dropping privileges to a `alpha` user who has a primary group of `alpha` and a member of the `beta` group and that service had `pkg_svc_group=beta`, then any directories created under the svc directory would be `alpha:alpha` (not `alpha:beta` which would be preferred). A sticky bit (which affects the group ownership) on a parent directory keeps that same group ownerships for new child dirs (I think files as well). In that way, the svc group name would percolate down the child dirs

Reply:
Reading section 15.4.5 of The Linux Programming Interface says something different... that the sticky bit is a restricted deletion flag, meaning an unprivileged process can only delete a file if it has write permissions on the directory _and_ also owns the file in question.